src/backend/joanie/core/api/client.py

```diff
@@ -403,15 +388,28 @@ def abort(self, request, pk=None):  # pylint: disable=no-self-use, invalid-name
                 f'No order found with id "{pk}" owned by {username}.', status=404
             )
 
-        if order.state != enums.ORDER_STATE_PENDING:
-            return Response("Cannot abort a not pending order.", status=403)
+        if order.state == enums.ORDER_STATE_VALIDATED:
+            return Response("Cannot abort a validated order.", status=403)
```

Comment:
By reading this again, I wonder if a [`422`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) status code is no more relevant than `403`.